### ISSUE TITLE
examples/cursor.py gives RuntimeError on mac osx

### ISSUE THREAD

**Issue by ifmihai** [matplotlib/matplotlib#3708]
I ran cursor.py, downloaded from the matplotlib/examples/widgets web page.

I'm on mac osx 10.9.4, anaconda latest as of today, matplotlib 1.3.1

the output is this:

``` python

79-118-141-238:wigets ifmichael$ python cursor.py
Traceback (most recent call last):
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/artist.py", line 55, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/figure.py", line 1040, in draw
    self.canvas.draw_event(renderer)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/backend_bases.py", line 1693, in draw_event
    self.callbacks.process(s, event)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/cbook.py", line 527, in process
    proxy(*args, **kwargs)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/cbook.py", line 405, in __call__
    return mtd(*args, **kwargs)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/widgets.py", line 865, in clear
    self.background = self.canvas.copy_from_bbox(self.ax.bbox)
AttributeError: 'FigureCanvasMac' object has no attribute 'copy_from_bbox'
Traceback (most recent call last):
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/backend_bases.py", line 1834, in motion_notify_event
    self.callbacks.process(s, event)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/cbook.py", line 527, in process
    proxy(*args, **kwargs)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/cbook.py", line 405, in __call__
    return mtd(*args, **kwargs)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/widgets.py", line 892, in onmove
    self._update()
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/widgets.py", line 899, in _update
    self.ax.draw_artist(self.linev)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/axes.py", line 2098, in draw_artist
    a.draw(self._cachedRenderer)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/artist.py", line 55, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/lines.py", line 526, in draw
    gc = renderer.new_gc()
  File "/Users/ifmichael/anaconda/lib/python2.7/site-packages/matplotlib/backends/backend_macosx.py", line 97, in new_gc
    self.gc.save()
RuntimeError: CGContextRef is NULL

```

Maybe this issue was resolved in 1.4.0, but I cannot install it (see #3517)

**Comment by jradavenport** [comment]
Seems like this is a long-standing issue... is the only workaround to force people to use a different backend? (e.g.  u'Qt4Agg', u'TkAgg')

**Comment by dopplershift** [comment]
This is working fine for me on 2.1 (and also works on 2.0.2) with Python 3.6. I bet this was fixed with #6178.

Feel free to re-open if someone is able to reproduce.
